builder/swig/go.bzl

```diff
@@ -0,0 +1,174 @@
+            allow_files = True,
+        ),
+        "package_name": attr.string(
+            doc = "Package name override for generated Go package",
+        ),
+        "enable_cxx": attr.bool(
+            doc = "Enable SWIG C++ processing (default: False)",
+            default = False,
+        ),
+        "extra_args": attr.string_list(
+            doc = "Extra arguments to be passed to SWIG",
+        ),
+        "_swig": attr.label(
+            default = Label("@swig//:swig"),
+            allow_single_file = True,
+            executable = True,
+            cfg = "host",
+        ),
+    },
+)
+
+def swig_go(name, lib, package_name, shared_lib_name=None, tags=[], **kwargs):
+    swig_wrapper_name = name + "_swig"
+
+    _swig_go_wrapper(
+        name = swig_wrapper_name,
+        lib = lib,
+        package_name=package_name,
+        **kwargs,
+    )
+
+    if not shared_lib_name:
+        shared_lib_name = name
+
+    native.filegroup(
+        name = "source_file",
+        srcs = [swig_wrapper_name],
+        output_group = "go_source",
+    )
+
+    # TODO windows would use copy, maybe use an if statement or select to check
+    cmd = "cp $(SRCS)/{pkg_name}.go  $@".format(pkg_name=package_name)
+
+    # Copy the generated go file into an output file with the same name.
+    native.genrule(
+        name="go_wrapper_copier",
+        outs= [package_name + ".go"],
+        srcs =[":source_file"],
+        cmd_bash = cmd,
+        visibility = ["//visibility:public"],
+    )
+
+    def swig_cc_binary(shared_lib_filename):
+        native.cc_binary(
+            name = shared_lib_filename,
+            deps = [lib, swig_wrapper_name],
+            srcs = [swig_wrapper_name],
+            linkshared = True,
+            linkopts = select({
+                    "@vaticle_bazel_distribution//platform:is_windows": ["ntdll.lib", "secur32.lib", "crypt32.lib", "ncrypt.lib"],
+                    "@vaticle_bazel_distribution//platform:is_mac": ["-framework CoreFoundation", "-install_name go/{filename}".format(filename=shared_lib_filename)],
+                    "//conditions:default": [],
+                }),
+        )
+
+    swig_cc_binary("lib" + shared_lib_name + ".dylib")
+    swig_cc_binary("lib" + shared_lib_name + ".so")
+    swig_cc_binary(shared_lib_name + ".dll")
+
+    native.alias(
+        name = "lib" + shared_lib_name,
+        actual = select({
+            "@vaticle_bazel_distribution//platform:is_mac": ("lib" + shared_lib_name + ".dylib"),
+            "@vaticle_bazel_distribution//platform:is_linux": ("lib" + shared_lib_name + ".so"),
+            "@vaticle_bazel_distribution//platform:is_windows": (shared_lib_name + ".dll"),
+        })
+    )
+
+    go_library(
+        name = name,
+        srcs = [package_name + ".go"],
+        data = ["lib" + shared_lib_name],
+        importpath = package_name + "/go_wrapper",
+        visibility = ["//visibility:public"],
+        cgo=True,
```

Comment:
Spaces around `=` =)

Reply:
Looks like you thought that I wanted you to change all the other assignments to `lhs=rhs` instead of `lhs = rhs`, although we use `lhs = rhs` everywhere... Fix it for this `go_library`, please.